torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -465,6 +491,60 @@ def _init_reshard_after_forward(self):
                 "sharding_strategy only supports FULL_SHARD and SHARD_GRAD_OP right now."
             )
 
+    def _get_ignored_modules(
+        self,
+        _ignored_modules: Any,
+    ) -> Set[torch.nn.Module]:
+        """
+        Checks that ``_ignored_modules`` is an iterable of ``torch.nn.Module``
+        s without any :class:`FullyShardedDataParallel` instances and then
+        returns them and their children as a :class:`set`, excluding nested
+        :class:`FullyShardedDataParallel` instances.
+
+        We include the child modules to better match user intuition since
+        ignoring a module should ignore its child modules as well, and we
+        exclude :class:`FullyShardedDataParallel` instances since ``self`` may
+        be the intended root instance that manages them.
+        """
+        if _ignored_modules is None:
+            return set()
+        msg_prefix = "`ignored_modules` should be an iterable of " \
+            "`torch.nn.Module`s "
+        try:
+            ignored_root_modules = set(_ignored_modules)
+        except TypeError:
+            raise TypeError(msg_prefix + f"but got {type(_ignored_modules)}")
+        for module in ignored_root_modules:
+            if not isinstance(module, torch.nn.Module):
+                raise TypeError(
+                    msg_prefix + f"but got an iterable with {type(module)}"
+                )
+            if isinstance(module, FullyShardedDataParallel):
+                raise ValueError(
+                    "`ignored_modules` should not include FSDP modules"
+                )
+        # Include child modules and exclude nested FSDP modules
+        ignored_modules = set(
+            child for module in ignored_root_modules
+            for child in module.modules()
+            if not isinstance(child, FullyShardedDataParallel) and
```

Comment:
This shouldn't be an expected use case, do we absolutely need support for it do you think?

Reply:
Sounds good. I think we can disallow this. I will update the PR to error in this case.

Reply:
Discussed offline: will support this since not too much added complexity.